main/src/ca/uwaterloo/flix/language/phase/typer/ConstraintGen.scala

```diff
@@ -478,12 +478,8 @@ object ConstraintGen {
 
       case Expr.ExtMatch(exp, rules, loc) =>
         val (tpe, eff) = visitExp(exp)
-        val (caseTpes, tpes, effs) = rules.map(visitExtMatchRule).unzip3
-        val expectedRowType = caseTpes.foldRight(Type.mkSchemaRowEmpty(loc)) {
-          case ((name, patTpes), acc) => Type.mkSchemaRowExtend(name, Type.mkRelation(patTpes, loc), acc, loc)
-        }
-        val expectedSchemaType = Type.mkExtensible(expectedRowType, loc)
-        c.unifyType(tpe, expectedSchemaType, exp.loc)
+        val (patTypes, tpes, effs) = rules.map(visitExtMatchRule).unzip3
+        c.unifyAllTypes(tpe :: patTypes, loc.asSynthetic)
```

Comment:
Two problems with this approach:
1. There is no way to enforce closed rows.
2. The scrutinee may not appear in any pattern but unify succesfully anyway.

I need to rethink this a bit.

Reply:
Example of problem 2:
```flix
ematch xvar A() {
    B() => ???
}
```